store/auth.go

```diff
@@ -175,14 +234,9 @@ func requestDischargeMacaroon(endpoint string, data map[string]string) (string,
 		return "", fmt.Errorf(errorPrefix+"server returned status %d", resp.StatusCode)
 	}
 
-	dec := json.NewDecoder(resp.Body)
-	var responseData struct {
-		Macaroon string `json:"discharge_macaroon"`
-	}
-	if err := dec.Decode(&responseData); err != nil {
+	if err != nil {
```

Comment:
looking at the full file there's also another place that I don't see in the diff with a repeated check like this